x-pack/test/functional/services/ml/data_frame_analytics_edit.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { MlCommon } from './common';
+
+export function MachineLearningDataFrameAnalyticsEditProvider(
+  { getService }: FtrProviderContext,
+  mlCommon: MlCommon
+) {
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+
+  return {
+    async assertJobDescriptionEditInputExists() {
+      await testSubjects.existOrFail('mlAnalyticsEditFlyoutDescriptionInput');
+    },
+    async assertJobDescriptionEditValue(expectedValue: string) {
+      const actualJobDescription = await testSubjects.getAttribute(
+        'mlAnalyticsEditFlyoutDescriptionInput',
+        'value'
+      );
+      expect(actualJobDescription).to.eql(
+        expectedValue,
+        `Job description edit should be '${expectedValue}' (got '${actualJobDescription}')`
+      );
+    },
+    async assertJobMmlEditInputExists() {
+      await testSubjects.existOrFail('mlAnalyticsEditFlyoutmodelMemoryLimitInput');
+    },
+    async assertJobMmlEditValue(expectedValue: string) {
+      const actualMml = await testSubjects.getAttribute(
+        'mlAnalyticsEditFlyoutmodelMemoryLimitInput',
+        'value'
+      );
+      expect(actualMml).to.eql(
+        expectedValue,
+        `Job model memory limit edit should be '${expectedValue}' (got '${actualMml}')`
+      );
+    },
+    async setJobDescriptionEdit(jobDescription: string) {
+      await mlCommon.setValueWithChecks('mlAnalyticsEditFlyoutDescriptionInput', jobDescription, {
+        clearWithKeyboard: true,
+      });
+      await this.assertJobDescriptionEditValue(jobDescription);
+    },
+
+    async setJobMmlEdit(mml: string) {
+      await mlCommon.setValueWithChecks('mlAnalyticsEditFlyoutmodelMemoryLimitInput', mml, {
+        clearWithKeyboard: true,
+      });
+      await this.assertJobMmlEditValue(mml);
+    },
+
+    async assertAnalyticsEditFlyoutMissing() {
+      await testSubjects.missingOrFail('analyticsEditFlyoutUpdateButton');
```

Comment:
The method name suggests that we're checking for a missing flyout (which is what we want to do here), but the check actually just validates that the button is missing. I'd suggest to check for `analyticsEditFlyout` (or for the renamed version, see other comment) instead.

Reply:
Good catch - fixed in 58d750b4268a3b164eda8f2b8cd3ab05160f9379